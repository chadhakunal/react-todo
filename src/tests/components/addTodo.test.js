import { shallow, mount } from "enzyme";
import AddTodo from "components/addTodo";

const clickFn = jest.fn();

describe('AddTodo', () => {
  it('should render correctly with no props', () => {
    const component = shallow(< AddTodo />);
    expect(component).toMatchSnapshot();
  });

  it('should call onAdd prop when add button is clicked', () => {
    const component = mount(< AddTodo onAdd={clickFn} />);
    const titleInput = component.find('input#title-input');
    const descriptionInput = component.find('textarea#description-input');
    titleInput.simulate('change', { target: { value: 'Hello' } });
    descriptionInput.simulate('change', { target: { value: 'World' } })
    
    component.find('button#add').simulate('click');
    
    expect(clickFn).toHaveBeenCalled();
    expect(titleInput.get(0).props.value).toEqual("");
    expect(descriptionInput.get(0).props.value).toEqual("");
    expect(descriptionInput.get(0).props.style).toHaveProperty('display', 'none');
    component.unmount();
  });

  it('should not call onAdd when title is missing', () => {
    const component = mount(< AddTodo onAdd={clickFn} />);
    component.find('button#add').simulate('click');
    expect(clickFn).not.toHaveBeenCalled();
    component.unmount();
  })

  it('should display description box when title is not empty', () => {
    const component = mount(< AddTodo onAdd={clickFn} />);
    const titleInput = component.find('input#title-input');
    const descriptionInput = component.find('textarea#description-input');
    
    titleInput.instance().value = "Hello"
    titleInput.simulate('change');
    
    expect(descriptionInput.instance().style).toHaveProperty('display', 'block');
    component.unmount();
  });

  it('should not display description box when title is empty', ()=>{
    const component = mount(< AddTodo onAdd={clickFn} />);
    const titleInput = component.find('input#title-input');
    const descriptionInput = component.find('textarea#description-input');
    expect(descriptionInput.instance().style).toHaveProperty('display', 'none');

    titleInput.instance().value = ""
    titleInput.simulate('change');
    
    expect(descriptionInput.instance().style).toHaveProperty('display', 'none');
    component.unmount();
  });
})
