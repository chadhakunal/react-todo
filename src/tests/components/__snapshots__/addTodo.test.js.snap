// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo should render correctly with no props 1`] = `
<div
  className="row w-100 pl-5 pr-5 pt-3 pb-3 justify-content-center"
>
  <InputGroup
    className="mt-2 ml-2 w-100 mb-3"
    tag="div"
  >
    <Input
      id="title-input"
      onChange={[Function]}
      placeholder="What would you like to do today?"
      type="text"
      value=""
    />
    <InputGroupAddon
      addonType="append"
      tag="div"
    >
      <Button
        className="add-button pl-4 pr-4"
        color="secondary"
        id="add"
        onClick={[Function]}
        tag="button"
      >
        Add
      </Button>
    </InputGroupAddon>
  </InputGroup>
  <Input
    className="description-input"
    id="description-input"
    onChange={[Function]}
    placeholder="Description (optional)"
    style={
      Object {
        "display": "none",
      }
    }
    type="textarea"
    value=""
  />
</div>
`;
